src/watertap_contrib/reflo/unit_models/crystallizer_effect.py

```diff
@@ -0,0 +1,430 @@
+
+from copy import deepcopy
+
+# Import Pyomo libraries
+from pyomo.environ import (
+    Var,
+    Param,
+    check_optimal_termination,
+    units as pyunits,
+)
+from pyomo.common.config import ConfigValue
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    useDefault,
+)
+from idaes.core.util.config import is_physical_parameter_block
+from idaes.core.util.exceptions import InitializationError
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslog
+
+from watertap.core.solvers import get_solver
+from watertap.core.util.initialization import interval_initializer
+from watertap.unit_models.crystallizer import CrystallizationData
+from watertap.unit_models.mvc.components.lmtd_chen_callback import (
+    delta_temperature_chen_callback,
+)
+
+from watertap_contrib.reflo.costing.units.multi_effect_crystallizer import (
+    cost_multi_effect_crystallizer,
+)
+
+__author__ = "Oluwamayowa Amusat, Zhuoran Zhang, Kurban Sitterley"
+
+
+@declare_process_block_class("CrystallizerEffect")
+class CrystallizerEffectData(CrystallizationData):
+    """
+    Zero dimensional model for crystallizer effect
+    """
+
+    CONFIG = CrystallizationData.CONFIG()
+
+    CONFIG.declare(
+        "property_package_vapor",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for heating steam properties",
+            doc="""Property parameter object used to define steam property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "standalone",
+        ConfigValue(
+            default=True,
+            domain=bool,
+            description="Flag to indicate if model is used alone or as part of MultiEffectCrystallizer.",
+            doc="""Flag to indicate if model is used alone or as part of MultiEffectCrystallizer unit model,
+    **default** - True.""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = False
+
+        self.properties_pure_water = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of pure water vapour at outlet",
+            **tmp_dict,
+        )
+
+        self.add_port(name="pure_water", block=self.properties_pure_water)
+
+        self.properties_pure_water[0].flow_mass_phase_comp["Liq", "H2O"].fix(1e-8)
+        self.properties_pure_water[0].flow_mass_phase_comp["Liq", "NaCl"].fix(0)
+        self.properties_pure_water[0].flow_mass_phase_comp["Sol", "NaCl"].fix(0)
```

Comment:
Is there a reason why water is 1e-8 and nacl components set to 0?

Reply:
This would be a question for @Zhuoran29

Reply:
I also believe there was some discussion of this in watertap-org/watertap#1252

Reply:
This was modified to fix the Vap water phase in ```properties_pure_water``` to zero.